examples/compiler/iet/tut_1.ipynb

```diff
@@ -0,0 +1,1162 @@
+     ]
+    }
+   ],
+   "source": [
+    "print(op.body[0].body[2])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Within this second list, the first element is responsible for the profiling procedure, while the second part – which is also a list (with only one element) – contains the kernel's main loop."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 48,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "<WithProperties[affine,sequential]::Iteration time[t0,t1]; (time_m, time_M, 1)>"
+      ]
+     },
+     "execution_count": 48,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "t_iter = op.body[0].body[2].body[1].body[0]\n",
+    "t_iter"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Here, we have captured the specific loop corresponding to the `time` dimension of our domain."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 53,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "for (int time = time_m, t0 = (time)%(2), t1 = (time + 1)%(2); time <= time_M; time += 1, t0 = (time)%(2), t1 = (time + 1)%(2))\n",
+      "{\n",
+      "  for (int x = x_m; x <= x_M; x += 1)\n",
+      "  {\n",
+      "    #pragma omp simd\n",
+      "    for (int y = y_m; y <= y_M; y += 1)\n",
+      "    {\n",
+      "      u[t1][x + 1][y + 1] = u[t0][x + 1][y + 1] + 1;\n",
+      "    }\n",
+      "  }\n",
+      "}\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(t_iter)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    " We can further investigate its limits, for instance."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 84,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "(time_m, time_M, 1)"
+      ]
+     },
+     "execution_count": 84,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "t_iter.limits"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "And as we keep going down through the IET, we reach the expression that is wrapped by the iterations' scope.  "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 89,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "<Expression u[t1, x + 1, y + 1] = u[t0, x + 1, y + 1] + 1>\n"
+     ]
+    }
+   ],
+   "source": [
+    "expr = op.body[0].body[2].body[1].body[0].children[0][0].body[0].children[0][0].children[0][0].body[0]\n",
+    "print(expr.view)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Devito already implements a default solution to reach all the expressions one can find in the IET of an operator."
```

Comment:
The compiler provides several IET visitors. One of these, ``FindNodes``, can be used to retrieve all nodes of a particular type within a given subtree. For example, we could retrieve all ``Expressions`` within an Operator